src/libraries/Common/tests/CoreFx.Private.TestUtilities/System/PlatformDetection.cs

```diff
@@ -150,6 +151,35 @@ public static string GetDistroVersionString()
             }
         }
 
+        private static Lazy<Version> m_icuVersion = new Lazy<Version>(GetICUVersion);
+        public static Version ICUVersion => m_icuVersion.Value;
+
+        public static bool ShouldUseIcu => ICUVersion > new Version(0,0,0,0);
+        public static bool ShouldUseNls => !ShouldUseIcu;
+
+        private static Version GetICUVersion()
```

Comment:
> Or does GlobalizationNative_GetICUVersion() also work (and return 0) on Win7/Win8?

It will return 0 if ICU is not loaded, that's the reason why I'm adding the `u_getVersion_ptr != NULL` check in `GlobalizationNative_GetICUVersion`. That is how we're deciding whether we should run tests on NLS or ICU.